<launch>
    <!-- PERCEPTION -->
    <node name="image_view_hsv" pkg="image_view" type="image_view" respawn="false" output="log">
        <remap from="image" to="/hsv_detector/image_result"/>
        <param name="autosize" value="true" />
    </node>

    <!-- <node name="image_view_hsv2" pkg="image_view" type="image_view" respawn="false" output="log">
        <remap from="image" to="/hsv_detector/image_result_thres"/>
        <param name="autosize" value="true" />
    </node> -->

    <param name="hsv_detector/reference_frame" value="/torso"/>
    <param name="hsv_detector/camera_frame"    value="/right_hand_camera"/>
    <param name="hsv_detector/area_threshold"  value="250"/>

    <node pkg="baxter_collaboration" type="hsv_detector" name="hsv_detector" output="screen">
        <remap from="hsv_detector/image" to="/cameras/right_hand_camera/image"/>
        <remap from="hsv_detector/camera_info" to="/cameras/right_hand_camera/camera_info"/>
    </node>

    <!-- Objects database for the left arm -->
    <rosparam param = "action_provider/objects_left">
        "table_top"   : 200
    </rosparam>

    <!-- Objects database for the right arm -->
    <rosparam param = "action_provider/objects_right">
        "screwdriver" : 1
        "blue_box"    : 2
    </rosparam>

    <!-- Objects database for the hsv cartesian detector -->
    <rosparam param = "hsv_detector/objects_db">
        [["screwdriver", [0.099, 0.030], [160,  10], [70, 166], [10,  66]],
         ["blue_box"   , [0.131, 0.131], [ 60, 130], [90, 256], [10, 256]]]
    </rosparam>

    <!-- CONTROL -->
    <!-- If the system should use the robot or not (useful for debugging) -->
    <arg name="useRobot"            default="true"/>

    <param name="action_provider/use_robot"             value="$(arg useRobot)"/>

    <!-- Parameters for force interaction -->
    <arg name="forceFilterVarianceL" default="0.001"/>
    <arg name="forceFilterVarianceR" default="1.0"/>
    <arg name="relativeForceThresholdL"   default="65.0"/>
    <arg name="relativeForceThresholdR"   default="1.0"/>

    <param name="action_provider/force_filter_variance_left"        value="$(arg forceFilterVarianceL)"/>
    <param name="action_provider/force_filter_variance_right"       value="$(arg forceFilterVarianceR)"/>
    <param name="action_provider/relative_force_threshold_left"     value="$(arg relativeForceThresholdL)"/>
    <param name="action_provider/relative_force_threshold_right"    value="$(arg relativeForceThresholdR)"/>

    <node pkg="baxter_collaboration" type="modular_action_provider" name="modular_action_provider" output="screen">
    </node>
</launch>

